docs/src/gui/qtvcp-widgets.adoc

```diff
@@ -75,543 +96,676 @@ LED #name_0f_led{
-The original widget must be in a layout for switcher to put it back.
+The _original widget must be in a layout_ for switcher to put it back.
 
-In Designer you will add the widgetswitcher widget on screen.
-Right click the widgetswitcher and add a page,
-then populate it with widgets/layouts you wish to see in a default form.
-Then add as many pages as there are views to switch to.
-on each page add a layout widget.
-After adding the layout you must right click the widget switcher again
-and set the layout option.
-click on the widgetswitcher widget and then scroll to the bottom of the
-property editor.
-you are looking for the dynamic property 'widget_list'.
-double click the to the right of the widget_list property.
-A dialog will pop up allowing you to add the names of the widgets to move
-to the pages you added to the widgetswitcher.
+In Qt Designer you will:
 
-There are function calls to display specific widgets:
+//FIXME Add an entry about widget numbering
+* Add the `WidgetSwitcher` widget on screen.
+* Right click the `WidgetSwitcher` and add a page.
+* Populate it with the widgets/layouts you wish to see in a default form.
+* Add as many pages as there are views to switch to.
+* On each page, add a layout widget. +
+  After adding the layout you must right click the widget switcher again
+  and set the layout option.
+* Click on the `WidgetSwitcher` widget and then scroll to the bottom of
+  the property editor.
+* Look for the dynamic property `widget_list` and double click to the
+  right of it.
+* A dialog pops up allowing you to add the names of the widgets to move
+  to the pages you added to the `WidgetSwitcher`.
 
-* [WidgetSwitcherName].show_id_widget(number)
-* [WidgetSwitcherName].show_named_widget(widget_name)
-* [WidgetSwitcherName].show_default()
-* [WidgetSwitcherName].show_next()
+There are _function calls_ to display specific widgets. +
+By calling one of these functions, you control what widget is currently
+displayed:
 
-By calling one of these functions, you control what widget
-is currently displayed. show_default() shows the page 0
-layout, and puts all other widgets back to where they were as initially
-built in Designer.
+*`[_WidgetSwitcherName_].show_id_widget(_number_)`*::
+*`[_WidgetSwitcherName_].show_named_widget(_widget_name_)`*::
+*`[_WidgetSwitcherName_].show_default()`*::
+  This shows the `page 0` layout, and puts all other widgets back to
+  where they were as initially built in Qt Designer.
+*`[_WidgetSwitcherName_].show_next()`*::
+  Show next widget.
 
+It is based on the _QStack_ widget.
 
-It is based on the QStack widget.
 
+[[sec:qtvcp:widgets:machine-controller]]
 == Machine Controller Widgets
 
-These widgets interact to the Machine Controller state.
+These widgets *interact with the Machine Controller state*.
 
-=== Action Button Widget
+[[sub:qtvcp:widgets:actionbutton]]
+=== `ActionButton` Controller Action Control Widget
```

Comment:
In my opinion all the headings are too long. This makes the table of contents appearing more as block of text than a list for getting an overview.
I would keep the headings short and add the correct name and the description inside the appropriate section.

![grafik](https://user-images.githubusercontent.com/67957916/189980583-2274f3d2-1e0f-403d-a445-42833158d6ae.png)

Reply:
As the desciptive part of these titles is already in first line of all these sections, those could be shortened to the bare widget name.
The new reader may in turn have to dig a little deeper to grasp the purpose of each widget... but the Fine Manual thou shall read ! ;)

Reply:
A separator like a dash may be helpful. I just find them borerlinish-long, not to long. But yes, this is beyond "optics only".

Reply:
The added dash indeed helps a bit for readability.

Reply:
Just a stupid question - what should the part after the dash say to me?
Like "Machine Controller Action Control Widget"...
Is it something that has to be used in the code when using these widgets?

Reply:
If it's a description then one more reason to shift it inside the section and away from the title IMHO.
But it's readable now, so we can keep it.

Reply:
And like I said, this small description is already on the first line of each section...

@c-morley what about stripping out the small desc from titles ? Are you OK with that ?